nvflare/private/fed/server/fed_server.py

```diff
@@ -1049,42 +1043,9 @@ def overseer_callback(self, overseer_agent):
 
     def _turn_to_hot(self):
         # Restore Snapshot
-        if self.ha_mode:
-            restored_job_ids = []
-            with self.snapshot_lock:
-                fl_snapshot = self.snapshot_persistor.retrieve()
-                if fl_snapshot:
-                    for run_number, snapshot in fl_snapshot.run_snapshots.items():
-                        if snapshot and not snapshot.completed:
-                            # Restore the workspace
-                            workspace_data = snapshot.get_component_snapshot(SnapshotKey.WORKSPACE).get("content")
-                            ws = Workspace(self.workspace)
-                            dst = ws.get_run_dir(str(run_number))
-                            if os.path.exists(dst):
-                                shutil.rmtree(dst, ignore_errors=True)
-
-                            os.makedirs(dst, exist_ok=True)
-                            unzip_all_from_bytes(workspace_data, dst)
-
-                            job_id = snapshot.get_component_snapshot(SnapshotKey.JOB_INFO).get(SnapshotKey.JOB_ID)
-                            job_clients = snapshot.get_component_snapshot(SnapshotKey.JOB_INFO).get(
-                                SnapshotKey.JOB_CLIENTS
-                            )
-                            self.logger.info(f"Restore the previous snapshot. Run_number: {run_number}")
-                            with self.engine.new_context() as fl_ctx:
-                                self.engine.job_runner.restore_running_job(
-                                    job_id=job_id,
-                                    job_clients=job_clients,
-                                    snapshot=snapshot,
-                                    fl_ctx=fl_ctx,
-                                )
-                            restored_job_ids.append(job_id)
-            with self.engine.new_context() as fl_ctx:
-                self.engine.job_runner.update_abnormal_finished_jobs(restored_job_ids, fl_ctx=fl_ctx)
-        else:
-            with self.engine.new_context() as fl_ctx:
-                self.snapshot_persistor.delete()
-                self.engine.job_runner.update_unfinished_jobs(fl_ctx=fl_ctx)
+        with self.engine.new_context() as fl_ctx:
+            self.snapshot_persistor.delete()
```

Comment:
next phase should we remove snapshot persistor? it might need major rework anyway in K8s so just leave it to you

Reply:
This is unknown for now.  Removing existing HA mechanism is one thing.  Leveraging K8s for HA is another.  However, K8s does not know what to restore.  The application has to keep its states in a persistent storage somewhere for later recovery.